### PR TITLE
Revamped flaming eyes, made effect_tindrift

### PR DESCRIPTION
**Pull request by Testudo-Cultu**
#### Summary

SUMMARY: Features "Flaming eyes have their own unique attack, Tindalos rift is JSON-ized "

For more on the meaning of each category, see:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md

#### Purpose of change

Flaming eyes causing teleglow immediately upon seeing the player was little more than a tedious anti-fungal tax than a real danger, especially later on. Tindalos rifts were hardcoded into teleglow, which made it practically impossible to use it on its own. 

#### Describe the solution

Flaming eyes now cause insanity, flavor-texted as "taint" to the player, causing constant hallucinations while debilitating the character. Effect gets stronger the longer you're seen, and at high intensity levels will begin creating Hounds of Tindalos rifts. This is by no means a nerf to flaming eyes. The intention is to make them genuinely challenging without lasers. 

#### Describe alternatives you've considered

The max severity level for the effect I added could be higher. I only kept it so low so it would be easier to implement as an attack by myself and others.

#### Testing

Spawned a flaming eye and got mauled by hounds. Effect intensity raises quickly, but tends to offer a chance to get out of sight, and all inflicted effects are removed properly. Teleporters still make you mushroomy sometimes.

#### Additional context

I have no idea if this fits into the lore of what flaming eyes are or what they're capable of, so if anyone knows, feel free to edit this or tell me what's wrong. This is my biggest PR yet and I'm sorry if I messed anything up with the formatting or just generally didn't do things like they're supposed to be done. I'm still learning a bit, but I'm glad to be able to help this game grow.

### CODE REVIEW

File: data/json/effects.json [hunk 1]
```diff
@@ -862,6 +862,36 @@
...
+  {
+    "type": "effect_type",
+    "id": "taint",
+    "name": [ "Touched mind", "Touched mind", "Tainted mind", "Badly tainted mind" ],
+    "miss_messages": [ [ "Your sense of reality warps!", 3 ] ],
+    "speed_name": "Tainted",
+    "desc": [
+      "You are disoriented as strange visions flash through your mind.",
+      "You are overwhelmed by the disturbing imagery and concepts you're flooded with.",
+      "You can't comprehend the things around you...",
```
Comment (Night-Pryanik) [inline data/json/effects.json:874]:
```suggestion
      "You can't comprehend the things around you…",
```

File: data/json/effects.json [hunk 2]
```diff
@@ -862,6 +862,36 @@
...
+      "You don't know what is and isn't real anymore..."
```
Comment (Night-Pryanik) [inline data/json/effects.json:875]:
```suggestion
      "You don't know what is and isn't real anymore…"
```

File: src/monattack.cpp [hunk 3]
```diff
@@ -116,7 +116,9 @@ static const efftype_id effect_shrieking( "shrieking" );
 static const efftype_id effect_slimed( "slimed" );
 static const efftype_id effect_stunned( "stunned" );
 static const efftype_id effect_targeted( "targeted" );
-static const efftype_id effect_teleglow( "teleglow" );
+static const efftype_id effect_tindrift( "tindrift" );
+static const efftype_id effect_taint( "taint" );
+static const efftype_id effect_hallu( "hallu" );
```
Comment (Night-Pryanik) [inline src/monattack.cpp:121]: Please put this into alphabetic order.

File: src/monattack.cpp [hunk 4]
```diff
@@ -2859,11 +2861,28 @@ bool mattack::stare( monster *z )
         if( g->u.sees( *z ) ) {
             add_msg( m_bad, _( "The %s stares at you, and you shudder." ), z->name() );
         } else {
-            add_msg( m_bad, _( "You feel like you're being watched, it makes you sick." ) );
+            add_msg( m_bad, _( "You feel like you're being watched; it makes you sick." ) );
```
Comment (Night-Pryanik) [inline src/monattack.cpp:2864]: This is a very minor change, and yet it will break previous translation. Please revert.

File: src/monattack.cpp [hunk 5]
```diff
@@ -2859,11 +2861,28 @@ bool mattack::stare( monster *z )
...
+        }
+       //Apply ?taint" to the target
```
Comment (Night-Pryanik) [inline src/monattack.cpp:2866]:
```suggestion
       //Apply "taint" to the target
```
BTW this is rather useless comment, the code below is easy to understand even without it.

File: src/monattack.cpp [hunk 6]
```diff
@@ -2859,11 +2861,28 @@ bool mattack::stare( monster *z )
...
+        g->u.add_effect( effect_taint, rng( 20_minutes, 60_minutes ) );
+       //Check severity before adding more debuffs
+                if ( g->u.get_effect_int ( effect_taint ) > 2 ) {
+        g->u.add_effect( effect_hallu, 30_minutes );
+        //Check if target is a player before spawning hallucinations
+        if( g->u.is_player() && one_in( 2 ) ) {
+            g->spawn_hallucination( g->u.pos() + tripoint( rng( -10, 10 ), rng( -10, 10 ), 0 ) );
+//        sound_hallu();
+        //add_msg( m_bad, _( "A strange figure appears!" ) );
```
Comment (Night-Pryanik) [inline src/monattack.cpp:2875]: Please remove commented out code.

**Comment by enaantd** [comment]
> I have no idea if this fits into the lore of what flaming eyes are or what they're capable of

I'm no expert about the lore, but I think flaming eyes becoming sentries from another world capable of sending alien hounds chasing you as opposed to magically making you blind even behind reinforced glass in a welcome change

File: src/monattack.cpp (hunk 5 above)
Comment (Testudo-Cultu) [reply src/monattack.cpp:2866]: It was only added in case I added more to the code. I just didn't think to remove it.

**Comment by Testudo-Cultu** [comment]
> 
> 
> > I have no idea if this fits into the lore of what flaming eyes are or what they're capable of
> 
> I'm no expert about the lore, but I think flaming eyes becoming sentinels from another world capable of sending alien hounds chasing you as opposed to magically making you blind even behind reinforced glass is a welcome change

With the changes I added they are still capable of making you blind, but only when the effect they cause is severe enough and only when in direct line of sight. The idea is that when they stare at you long enough, you begin to see otherworldly and disturbing things that overtake your vision entirely.

**Comment by enaantd** [comment]
> With the changes I added they are still capable of making you blind, but only when the effect they cause is severe enough and only when in direct line of sight. The idea is that when they stare at you long enough, you begin to see otherworldly and disturbing things that overtake your vision entirely.

Seems fair enough to me, as it's not "become instantly blind upon seeing a flaming eye"

**Comment by Inglonias** [comment]
Where does this leave the tinfoil hat? Tinfoil hats could be used to mitigate the effects of flaming eyes and now that I type that out, it seems very silly. Carry on.

**Comment by Testudo-Cultu** [comment]
> 
> 
> Where does this leave the tinfoil hat? Tinfoil hats could be used to mitigate the effects of flaming eyes and now that I type that out, it seems very silly. Carry on.

Actually, there is a tag for psychic shielding called PSYSHIELD_PARTIAL that's also used by Amigara horrors in-game. I considered making this tag prevent the effects of flaming eyes, but I decided not to. My idea was that the flaming eyes' hallucinations are less of a psionic attack and moreso that they warp reality, so dimensional anchors are the only way to counter it.

**Comment by Inglonias** [comment]
@Testudo-Cultu As I said, typing out the fact that tinfoil hats help you sort of feels silly cobsidering the direction things have been headed in. I personally agree with you that flaming eyes causing random fungal infections was kind of infuriating, and am glad to see an alternate proposal

**Comment by Testudo-Cultu** [comment]
> 
> 
> Please remove unrelated changes (specifically, in `effects.json`).
> Please astyle your changes.

I deleted the duplicate effects.json and I used OTBS as the bracket style. Could you please tell me (if there are any) what the other parameters are for astyle?

**Comment by Night-Pryanik** [comment]
> Could you please tell me (if there are any) what the other parameters are for astyle?

The full list of astyle rules we use in the project are described in `.astylerc` file.

File: src/player_hardcoded_effects.cpp [hunk 7]
```diff
@@ -644,29 +655,24 @@ void player::hardcoded_effects( effect &it )
...
-                fall_asleep( 2_hours );
-                if( one_in( 6 ) ) {
-                    // Set ourselves up for removal
-                    it.set_duration( 0_turns );
-                }
+        }
+        if( one_in( 7200 - ( ( dur - 600_minutes ) / 30_seconds ) ) && one_in( 20 ) ) {
+            if( !is_npc() ) {
+                add_msg( m_bad, _( "You pass out." ) );
+            }
```
Comment (Night-Pryanik) [inline src/player_hardcoded_effects.cpp:671]:
```suggestion
            if( !is_npc() ) {
                add_msg( m_bad, _( "You pass out." ) );
            }
```
```suggestion
            add_msg_if_player( m_bad, _( "You pass out." ) );
```

Comment (Testudo-Cultu) [reply src/player_hardcoded_effects.cpp]: That bit of code wasn't my doing, but if necessary, please let me know if there's anything else that should be cleaned up.

**Pull request merged**